File: LeetCode/9_palindrome_number.ts
// Palindrome Number
// https://leetcode.com/problems/palindrome-number/

function isPalindrome(x: number): boolean {
  if (String(x)[0] === '-') return false;

  for (let i = 0; i < String(x).length; i++) {
    if (String(x)[i] !== String(x)[String(x).length - i - 1]) {
      return false;
    }
  }

  return true;
}

File: LeetCode/9_palindrome_number.py
#  9. Palindrome Number
#  https://leetcode.com/problems/palindrome-number/

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0: return False
        if 0 < x <= 9: return True

        original = x
        num = x
        reverse = 0

        for i in str(x):
            tmp = num % 10
            num = num // 10

            reverse = reverse * 10 + tmp

        return reverse == original
